ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,529 @@
+	response, detail, err := sess.GetMonitor(getMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Monitor:%s\n%s", err, detail)
+	}
+
+	d.Set("id", response.ID)
+	d.Set(pdnsInstanceID, idset[0])
+	d.Set(pdnsGlbMonitorID, response.ID)
+	d.Set(pdnsGlbMonitorName, response.Name)
+	d.Set(pdnsGlbMonitorCreatedOn, response.CreatedOn)
+	d.Set(pdnsGlbMonitorModifiedOn, response.ModifiedOn)
+	d.Set(pdnsGlbMonitorType, response.Type)
+	d.Set(pdnsGlbMonitorPort, response.Port)
+	d.Set(pdnsGlbMonitorInterval, response.Interval)
+	d.Set(pdnsGlbMonitorRetries, response.Retries)
+	d.Set(pdnsGlbMonitorTimeout, response.Timeout)
+	if response.Description != nil {
+		d.Set(pdnsGlbMonitorDescription, response.Description)
+	}
+	if *response.Type == "HTTP" || *response.Type == "HTTPS" {
+		d.Set(pdnsGlbMonitorMethod, response.Method)
+		d.Set(pdnsGlbMonitorPath, response.Path)
+		d.Set(pdnsGlbMonitorExpectedCodes, response.ExpectedCodes)
+		d.Set(pdnsGlbMonitorHeaders, flattenDataSourceLoadBalancerHeader(response.HeadersVar))
+		if response.ExpectedBody != nil {
+			d.Set(pdnsGlbMonitorExpectedBody, response.ExpectedBody)
+		}
+	}
+	if *response.Type == "HTTPS" {
+		if response.AllowInsecure != nil {
+			d.Set(pdnsGlbMonitorAllowInsecure, response.AllowInsecure)
+		}
+	}
+	return nil
+}
+
+func flattenDataSourceLoadBalancerHeader(header []dnssvcsv1.HealthcheckHeader) interface{} {
+	flattened := make([]interface{}, 0)
+	for k, v := range header {
+		cfg := map[string]interface{}{
+			pdnsGlbMonitorHeadersName:  k,
+			pdnsGlbMonitorHeadersValue: v,
+		}
+		flattened = append(flattened, cfg)
+	}
+	return flattened
+}
+
+func resourceIBMPrivateDNSGLBMonitorUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	idset := strings.Split(d.Id(), "/")
+
+	// Update PDNS GLB Monitor if attributes has any change
+
+	if d.HasChange(pdnsGlbMonitorName) ||
+		d.HasChange(pdnsGlbMonitorDescription) ||
+		d.HasChange(pdnsGlbMonitorInterval) ||
+		d.HasChange(pdnsGlbMonitorRetries) ||
+		d.HasChange(pdnsGlbMonitorTimeout) ||
+		d.HasChange(pdnsGlbMonitorExpectedBody) {
+		updateMonitorOptions := sess.NewUpdateMonitorOptions(idset[0], idset[1])
+		uname := d.Get(pdnsGlbMonitorName).(string)
+		udescription := d.Get(pdnsGlbMonitorDescription).(string)
+		uinterval := int64(d.Get(pdnsGlbMonitorInterval).(int))
+		uretries := int64(d.Get(pdnsGlbMonitorRetries).(int))
+		utimeout := int64(d.Get(pdnsGlbMonitorTimeout).(int))
+		updateMonitorOptions.SetName(uname)
+		updateMonitorOptions.SetDescription(udescription)
+		updateMonitorOptions.SetInterval(uinterval)
+		updateMonitorOptions.SetRetries(uretries)
+		updateMonitorOptions.SetTimeout(utimeout)
+
+		_, detail, err := sess.UpdateMonitor(updateMonitorOptions)
+
+		if err != nil {
+			return fmt.Errorf("Error updating pdns GLB Monitor:%s\n%s", err, detail)
+		}
+	}
+
+	if d.HasChange(pdnsGlbMonitorType) ||
```

Comment:
keep this d.HasChange checks.. remove above one.

Reply:
done